instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py

```diff
@@ -342,24 +351,41 @@ def _populate_span(
         if self._db_api_integration.capture_parameters and len(args) > 1:
             span.set_attribute("db.statement.parameters", str(args[1]))
 
+    def get_operation_name(self, cursor, args):
+        if args and isinstance(args[0], str):
+            return args[0].split(" ")[0]
+        return ""
```

Comment:
> It is not recommended to attempt any client-side parsing of db.statement just to get these properties, they should only be used if the library being instrumented already provides them.

Spec explicitly says it doesn't recommend manipulating statement to get the properties. But I also agree that it has very high cardinal value without variable arguments. If we are going to parse statement may be it's worth considering something like [sqlparse](https://github.com/andialbrecht/sqlparse) and have structured way of making span name from tokens.

Reply:
I agree that we shouldn't parse it either but I don't think this really counts as "SQL parsing". We are making a bare-bones, extremely cheap, best effort attempt at guessing the SQL operation name. I think as long as it does not incur a perf penalty, decrease reliabilit, affect the DB connection/cursor/statement in any way or end up talking to the database, we should be good. At least that's how I understand the intention behind the spec recommendation but may be there is more to it that I might have overlooked?

Reply:
Yes, this isn't really a parsing. I thought the motive behind the spec recommendation is to not derive/guess anything from db statement. Otherwise this is good and it doesn't really cost anything in terms of perf/reliability etc... May be one other point worth bringing up here is different commands like `drop database ...`, `drop table ...`, `drop index ...` and `drop view ...` etc.. will have same span name and effectively making it a few set of keywords.

Reply:
And given that the most usual operations are mainly select, update, create we are good this change.

Reply:
Might be worth updating the spec to recommend this cheap "parsing"?

Reply:
Update: Given this discussion, the discussion on another PR (https://github.com/open-telemetry/opentelemetry-python-contrib/pull/254#discussion_r542925671) and another discussion that happened in the SIG meeting, we agreed that this does not constitute to SQL parsing and agreed to move forward with the solution.

Reply:
@aabmass good idea to clarify it in the spec. I'll try to submit a PR soon.